js/localization/ldml/date.parser.js

```diff
@@ -181,7 +200,9 @@ var getRegExpInfo = function(format, dateParts) {
             regexpText += "(" + regexpPart(count, dateParts) + ")";
             i += count - 1;
         } else {
-            regexpText += char;
+            char = escapeRegExp(char);
+            patterns.push(char);
```

Comment:
How you differ pattern "y" and separator "y" if patterns[i] for these cases are equal? Maybe, a better way to introduce a special pattern symbol for separators?

Reply:
You can escape this letter: `'year: 'yyyy`